subsys/bluetooth/mesh/chat_cli.c

```diff
@@ -0,0 +1,269 @@
+{
+	struct bt_mesh_chat_cli *chat = model->user_data;
+	struct bt_mesh_chat_cli_presence_status presence;
+
+	presence.presence = chat->presence;
+
+	(void)bt_mesh_chat_cli_presence_status_pub(chat, ctx, &presence);
+}
+
+/* .. include_startingpoint_chat_cli_rst_2 */
+const struct bt_mesh_model_op _bt_mesh_chat_cli_op[] = {
+	{ BT_MESH_CHAT_CLI_OP_MESSAGE,
+			BT_MESH_CHAT_CLI_MSG_MINLEN_MESSAGE,
+			handle_message },
+	{ BT_MESH_CHAT_CLI_OP_MESSAGE_REPLY,
+			BT_MESH_CHAT_CLI_MSG_LEN_MESSAGE_REPLY,
+			handle_message_reply },
+	{ BT_MESH_CHAT_CLI_OP_PRESENCE_STATUS,
+			BT_MESH_CHAT_CLI_MSG_LEN_PRESENCE_STATUS,
+			handle_presence_status },
+	{ BT_MESH_CHAT_CLI_OP_PRESENCE_GET,
+			BT_MESH_CHAT_CLI_MSG_LEN_PRESENCE_GET,
+			handle_presence_get },
+	BT_MESH_MODEL_OP_END,
+};
+/* .. include_endpoint_chat_cli_rst_2 */
+
+static int bt_mesh_chat_cli_update_handler(struct bt_mesh_model *model)
+{
+	struct bt_mesh_chat_cli *chat = model->user_data;
+
+	/* Continue publishing current presence. */
+	encode_presence(model->pub->msg, chat->presence);
+
+	return 0;
+}
+
+/* .. include_startingpoint_chat_cli_rst_3 */
+#ifdef CONFIG_BT_SETTINGS
+static int bt_mesh_chat_cli_settings_set(struct bt_mesh_model *model,
+					 const char *name,
+					 size_t len_rd,
+					 settings_read_cb read_cb,
+					 void *cb_arg)
+{
+	struct bt_mesh_chat_cli *chat = model->user_data;
+
+	if (name) {
+		return -ENOENT;
+	}
+
+	ssize_t bytes = read_cb(cb_arg, &chat->presence,
+				sizeof(chat->presence));
+	if (bytes < 0) {
+		return bytes;
+	}
+
+	if (bytes != 0 && bytes != sizeof(chat->presence)) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+#endif
+/* .. include_endpoint_chat_cli_rst_3 */
+
+/* .. include_startingpoint_chat_cli_rst_4 */
+static int bt_mesh_chat_cli_init(struct bt_mesh_model *model)
+{
+	struct bt_mesh_chat_cli *chat = model->user_data;
+
+	/* Store pointer to @ref bt_mesh_model for the later usage. */
+	chat->model = model;
+
+	/* Initialize model publication context. */
+	net_buf_simple_init_with_data(&chat->pub_msg, chat->buf,
+				      sizeof(chat->buf));
+	chat->pub.msg = &chat->pub_msg;
+	chat->pub.update = bt_mesh_chat_cli_update_handler;
+
+	/* Initialize ack message context. */
+	model_ack_init(&chat->ack_ctx);
+
+	return 0;
+}
+/* .. include_endpoint_chat_cli_rst_4 */
+
+const struct bt_mesh_model_cb _bt_mesh_chat_cli_cb = {
+	.init = bt_mesh_chat_cli_init,
+#ifdef CONFIG_BT_SETTINGS
+	.settings_set = bt_mesh_chat_cli_settings_set,
+#endif
+};
+
+/* .. include_startingpoint_chat_cli_rst_5 */
+int bt_mesh_chat_cli_presence_status_pub(struct bt_mesh_chat_cli *chat,
+			 struct bt_mesh_msg_ctx *ctx,
+			 const struct bt_mesh_chat_cli_presence_status *pres)
+{
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_CHAT_CLI_OP_PRESENCE_STATUS,
+				 BT_MESH_CHAT_CLI_MSG_LEN_PRESENCE_STATUS);
+
+	if (pres->presence != chat->presence) {
+		(void)bt_mesh_model_data_store(chat->model, true, NULL,
+					       &pres->presence,
+					       sizeof(chat->presence));
+	}
```

Comment:
This is an unexpected side effect of this function, based on the name. Perhaps `bt_mesh_chat_cli_presence_set` would be more appropriate?